### PR TITLE
Beam Website Updates for 2.58.0 Release

### PR DESCRIPTION
**Pull request by jrmccluskey**
Updates the Beam website with the 2.58.0 release blog post and download links.

DO NOT MERGE until 2.58.0 is approved 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: website/www/site/content/en/blog/beam-2.58.0.md
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:28]:
```suggestion
For more information about changes in 2.58.0, check out the [detailed release notes](https://github.com/apache/beam/milestone/22).
```

File: website/www/site/content/en/blog/beam-2.58.0.md
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:40]: Is it worth writing out LGPL, or will users know what this is?

Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:40]:
```suggestion
* Removed a third-party LGPL dependency from the Go SDK ([#31765](https://github.com/apache/beam/issues/31765)).
```

File: website/www/site/content/en/blog/beam-2.58.0.md
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:41]:
```suggestion
* Support for `MapState` and `SetState` when using Dataflow Runner v1 with Streaming Engine (Java) ([[#18200](https://github.com/apache/beam/issues/18200)])
```

File: website/www/site/content/en/blog/beam-2.58.0.md
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:45]:
```suggestion
* [IcebergIO] `IcebergCatalogConfig` was changed to support specifying catalog properties in a key-store fashion ([#31726](https://github.com/apache/beam/pull/31726))
```

File: website/www/site/content/en/blog/beam-2.58.0.md
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:47]:
```suggestion
## Bug fixes
```

**Review by rszper — APPROVED** [review]

File: website/www/site/content/en/blog/beam-2.58.0.md
Comment (jrmccluskey) [reply website/www/site/content/en/blog/beam-2.58.0.md]: LGPL should be fine here

File: CHANGES.md
Comment (tvalentyn) [inline CHANGES.md:94]: This sounds like a long-anticipated addition, so worth making a highlight.

File: CHANGES.md
Comment (tvalentyn) [inline CHANGES.md:111]:
```suggestion
* [BigQueryIO] Fixed a bug in batch Storage Write API that frequently exhausted concurrent connections quota ([#31710](https://github.com/apache/beam/pull/31710))
* Fixed a logging issue where Python worker dependency installation logs sometimes were not emitted in a timely manner ([#31977](https://github.com/apache/beam/pull/31977)).
```

File: CHANGES.md
Comment (tvalentyn) [reply CHANGES.md:94]: Thanks, let's reflect the changes here also in CHANGES.md on master after this is merged.

File: website/www/site/content/en/get-started/downloads.md
Comment (tvalentyn) [inline website/www/site/content/en/get-started/downloads.md:101]: use 2.58.0 (multiple places in URLs need a tweak)

Comment (jrmccluskey) [reply website/www/site/content/en/get-started/downloads.md]: fixed

File: website/www/site/content/en/blog/beam-2.58.0.md
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:24]: I assume this URL will be updated before we publish.

File: website/www/site/content/en/blog/beam-2.58.0.md
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:46]:
```suggestion
* [SpannerIO] Added validation that query and table cannot be specified at the same time for `SpannerIO.read()`. Previously `withQuery` overrides `withTable`, if set ([#24956](https://github.com/apache/beam/issues/24956)).
```

File: website/www/site/content/en/blog/beam-2.58.0.md
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:54]:
```suggestion
According to git shortlog, the following people contributed to the 2.58.0 release. Thank you to all contributors!
```

**Review by rszper — APPROVED** [review]

File: website/www/site/content/en/blog/beam-2.58.0.md
Comment (jrmccluskey) [reply website/www/site/content/en/blog/beam-2.58.0.md:24]: Yes the dates are going to be updated now that the RC has been approved

**Comment by jrmccluskey** [comment]
Going to check the links once the staging precommit is done, will merge after

**Pull request merged**
